src/model.h

```diff
@@ -225,6 +225,12 @@ class Model {
 
   uint64_t MaxPriorityLevel() const { return max_priority_level_; }
 
+  // Returns the scheduler's metric reporter
+  std::shared_ptr<MetricModelReporter> MetricReporter()
+  {
+    return scheduler_->MetricReporter();
```

Comment:
Should metric reporter be owned by the model?

Reply:
It think it makes a bit more sense to promote ownership from the scheduler to the model, and it will remove reporter from the scheduler interace. I'm not that opinionated on promoting it at this time, but I'll try it out.

Reply:
Promoted reporter to model, and realized I could more easily remove the the `response_cache_enable` that was getting passed around everywhere: https://github.com/triton-inference-server/core/pull/246/commits/b9527b5ba46a17fcd0d86fc51e653b452954e24e